src/mpid/ch4/src/ch4_progress.c

```diff
@@ -61,6 +61,14 @@ static int MPIDI_Progress_test(MPID_Progress_state * state)
     goto fn_exit;
 }
 
+int MPIDI_Progress_test(int flags)
+{
+    MPID_Progress_state state;
+    MPID_Progress_start(&state);
+    state.flag = flags;
+    return progress_test(&state);
```

Comment:
I just prefer pass `flags` as parameter of `MPID_Progress_start`.

Reply:
`MPID_Progress_start` is used by `MPIR_Wait_impl` and `MPIR_Waitall_impl`, nothing else should call it (unless they are internal usage)

Reply:
Note that we don't change ADI interface easily.